core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -452,21 +460,68 @@ object StorageTool extends Logging {
       stream.println("All of the log directories are already formatted.")
     } else {
       metaPropertiesEnsemble.emptyLogDirs().forEach(logDir => {
+        System.out.println("logDir : "+ logDir)
         copier.setLogDirProps(logDir, new MetaProperties.Builder(metaProperties).
           setDirectoryId(copier.generateValidDirectoryId()).
           build())
         copier.setPreWriteHandler((logDir, _, _) => {
           stream.println(s"Formatting $logDir with metadata.version $metadataVersion.")
           Files.createDirectories(Paths.get(logDir))
           val bootstrapDirectory = new BootstrapDirectory(logDir, Optional.empty())
+          System.out.println("bootstrapDirectory : "+ bootstrapDirectory)
           bootstrapDirectory.writeBinaryFile(bootstrapMetadata)
         })
         copier.setWriteErrorHandler((logDir, e) => {
           throw new TerseFailure(s"Error while writing meta.properties file $logDir: ${e.getMessage}")
         })
         copier.writeLogDirChanges()
+        // Write new file checkpoint file if standalone mode
+        if(standaloneMode){
+          writeCheckpointFile(logDir)
+        }
       })
     }
     0
   }
+
+  def writeCheckpointFile(logDir: String): Unit = {
+    val snapshotCheckpointDir = logDir + "/__cluster_metadata-0"
+
+    // Ensure the directory exists
+    val snapshotDir = Paths.get(snapshotCheckpointDir)
+    if (!Files.exists(snapshotDir)) {
+      Files.createDirectories(snapshotDir)
+    }
+
+    // Create the full path for the checkpoint file
+    val checkpointFilePath = snapshotDir.resolve(snapshotDir)
+
+    // Create the raw snapshot writer
+    val rawSnapshotWriter = FileRawSnapshotWriter.create(checkpointFilePath, new OffsetAndEpoch(0, 0))
+
+    val builder = new RecordsSnapshotWriter.Builder()
+      .setKraftVersion(1)
+      .setVoterSet(Optional.empty())
+      .setRawSnapshotWriter(rawSnapshotWriter).build(new MetadataRecordSerde)
+
+    // Append control records: KRaftVersionRecord and VotersRecord
+    appendControlRecords(builder)
```

Comment:
This actually write those records as "metadata" records and not as controller records.

You don't need to call this method as the `RecordsSnapshotWriter.Builder.build(RecordSerde)` will write the necessary control record if you configure the builder correctly by calling `setKraftVersion` and `setVoterSet`.